rocAL/source/decoders/video/rocdec_video_decoder.cpp

```diff
@@ -0,0 +1,184 @@
+    
+    VideoDecoder::Status status = Status::OK;
+    VideoSeekContext video_seek_ctx;
+
+    // Reconfig the decoder
+    ReconfigDumpFileStruct reconfig_user_struct = { 0 };
+    ReconfigParams reconfig_params = { 0 };
+    reconfig_params.p_fn_reconfigure_flush = ReconfigureFlushCallback;
+    reconfig_params.reconfig_flush_mode = RECONFIG_FLUSH_MODE_NONE;
+    reconfig_params.p_reconfig_user_struct = &reconfig_user_struct;
+    _rocvid_decoder->SetReconfigParams(&reconfig_params);
+
+    if (!_demuxer || !_rocvid_decoder || !out_buffer) {
+        ERR("Decoder is not initialized");
+        return Status::FAILED;        
+    }
+    if (!out_buffer || !(sequence_length|stride)) {
+        ERR("Invalid parameter passed");
+        return Status::FAILED;        
+    }
+
+    int64_t pts = 0, dts = 0, requested_frame_dts = 0;
+    int n_video_bytes = 0, n_frame_returned = 0, n_frame = 0, pkg_flags = 0;
+    OutputSurfaceInfo *surf_info;
+    uint8_t *pvideo = nullptr;
+    int num_decoded_frames = sequence_length * stride;
+
+    bool b_seek = true;       // seek only for first frame in sequence
+    uint32_t image_size = out_height * out_stride * sizeof(uint8_t);
+    video_seek_ctx.seek_crit_ = SEEK_CRITERIA_FRAME_NUM;
+    video_seek_ctx.seek_mode_ = SEEK_MODE_PREV_KEY_FRAME;
+    VideoPostProcess post_process;
+    bool sequence_decoded = false;
+    do {
+        if (b_seek) {
+            video_seek_ctx.seek_frame_ = static_cast<uint64_t>(seek_frame_number);
+            video_seek_ctx.seek_crit_ = SEEK_CRITERIA_FRAME_NUM;
+            video_seek_ctx.seek_mode_ = SEEK_MODE_PREV_KEY_FRAME;
+            _demuxer->Seek(video_seek_ctx, &pvideo, &n_video_bytes);
+            pts = video_seek_ctx.out_frame_pts_;
+            dts = video_seek_ctx.out_frame_dts_;
+            requested_frame_dts = video_seek_ctx.requested_frame_dts_;
+            b_seek = false;
+            _rocvid_decoder->FlushAndReconfigure();
+        } else {
+            _demuxer->Demux(&pvideo, &n_video_bytes, &pts, &dts);
+        }
+        // Treat 0 bitstream size as end of stream indicator
+        if (n_video_bytes == 0) {
+            pkg_flags |= ROCDEC_PKT_ENDOFSTREAM;
+        }
+        n_frame_returned = _rocvid_decoder->DecodeFrame(pvideo, n_video_bytes, pkg_flags, pts);
+        if (!n_frame && !_rocvid_decoder->GetOutputSurfaceInfo(&surf_info)) {
+            ERR("Failed to get Output Surface Info!");
+            return Status::FAILED;
+        }
+        // Take the min of sequence length and num frames to avoid out of bounds memory error
+        int required_n_frames = std::min(static_cast<int>(sequence_length), n_frame_returned);
+        for (int i = 0; i < required_n_frames; i++) {
+            uint8_t *pframe = _rocvid_decoder->GetFrame(&pts);
+            if (dts >= requested_frame_dts) {
```

Comment:
how is the dts value updated here. We are only returning pts in the call

Reply:
The dts value returned by the demuxer is used to go to the correct position and start decoding from that point